docs/arch/adr-03.md

```diff
@@ -0,0 +1,112 @@
+
+### Option 2: Azure File Service
+
+#### Specs
+Target throughput for a single file share: up to 300 MiB/sec for certain regions, Up to 60 MiB/sec for all regions
+
+#### Pros: 
+- can cache Azure file shares on on-premises file servers by using Azure File Sync for quick access
+- File Services uses the SMB protocol, which is the same protocol used on file directories on Mac and Windows machines. Therefore a file share can be mapped onto a drive on your machine. Note that a blob container can also be mounted as a file system using the [blobfuse](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-how-to-mount-container-linux) tool, so this is not an actual advantage over Blob Storage.
+- Greater potential throughput
+
+#### Cons:
+- Launched in Sept 2015, hasn't been around for as long as Blob Storage.
+- Smaller max size (100 TiB: 1 TiB = 2^40 bytes). We are not expecting to come close to this size, so this isn’t much of a concern
+
+## Decision
+
+We don't appear to have any need for most of the additional functionality that comes with File Service, which makes me reluctant to want to use it. In addition, the number of articles and resources available on communicating with Blob Storage to set up file zipping is much greater than what's available for File Service. My initial understanding of Blob Storage led me to believe that permissions could only be set at the container level, but this turned out to be wrong. With the ability to set blob-specific permissions, we will be able to use a single container to store the transcription-specific files, and the user-requested zip files.
+
+Ultimately, my choice is to go with Blob Storage: the more basic, simple storage tool that gives us what we need and nothing more. That being said, I'd still like to keep the option of using Azure File Service on the table, in case it turns out that we *would* benefit from the additional functionality that it offers.
+
+As for what type of blob we will use, my choice would be to store each data file in its own block blob. If we were to choose to store multiple files within a single blob (and have each file be associated with a block ID on that blob), we would lose the ability to name each individual file. Hypothetically, it would be possible to create a database table with columns “block ID” and “name”, to emulate a naming functionality, but this seems far more complicated than its worth. In addition, the [azure-storage-blob](https://github.com/azure/azure-storage-ruby/tree/master/blob) gem gives us a simple interface for working with block blobs and saves us the trouble of having to write HTTP requests ourselves.
```

Comment:
I don't disagree with your conclusion - rather my limited understanding about the append only blob is we treat it as one file / blob and pull data out of it, e.g. each row in the file/blob is one of our approved transcriptions.

Reply:
Hmm... I guess I don't see the benefit of storing data for multiple approved transcriptions in a single file, seems like it'd make it more complicated when we are grabbing the data from azure and organizing it into files that users are expecting to get. In our [design doc](https://docs.google.com/document/d/1erl1M72pNCkMBGxrqbds3vdquw5PxTSNQqfg7gMfhPw/edit?pli=1#) that delineates how the transcription data will be saved for export, it's been decided that there will be 3 separate files for each transcription:

> When a subject is Approved, a worker will read subject data from the db and write to several files:
>- Data table with metadata created in the app (db export, explicit fields TBD: .csv)
>- File with transcription only (plain text, line by line: .txt)
>- Raw, unparsed transcription (full jsonb: .json);

@camallen maybe there's something I'm missing from what you're saying - would it help to chat about it on a quick call?

Reply:
Sounds like you have this under control, no need for a call.